lib/Conversion/TorchOnnxToTorch/DefaultDomainGtoP.cpp

```diff
@@ -558,6 +558,95 @@ void mlir::torch::onnx_c::populateDefaultDomainGtoP(
                       binder.op, resultType, lhs, rhs);
                   return success();
                 });
+  patterns.onOp(
+      "GatherND", 13, [](OpBinder binder, ConversionPatternRewriter &rewriter) {
+        Torch::ValueTensorType resultType;
+        Value data, indices;
+        int64_t batch_dims;
+        if (binder.tensorOperandAtIndex(data, 0) ||
+            binder.tensorOperandAtIndex(indices, 1) ||
+            binder.tensorResultType(resultType) ||
+            binder.s64IntegerAttr(batch_dims, "batch_dims", 0))
+          return failure();
+        // TODO: Add support for non zero batch_dims.
```

Comment:
I would prefer a plan for how to support batch dimensions before adding this lowering. If it turns out that we cannot back to `onnx.Gather` with batch dims then we might as well not have this lowering for the non-batch dims case.

Reply:
yeah, to support batch_dims we can not map to onnx.gather op. Updated the lowering algorithm. Thanks!